### PR TITLE
d_menu_fishing mostly done

### PR DESCRIPTION
**Pull request by Trueffeloot**
Function setFishParam with a regalloc prevents the whole file from being finished

### CODE REVIEW

File: include/d/menu/d_menu_fishing.h [hunk 1]
```diff
@@ -38,10 +38,10 @@ class dMenu_Fishing_c : public dDlst_base_c {
 
 private:
     /* 0x004 */ JKRExpHeap* mpHeap;
-    /* 0x008 */ JKRArchive* field_0x8;
+    /* 0x008 */ JKRArchive* mpArchive;
     /* 0x00C */ STControl* mpStick;
     /* 0x010 */ CSTControl* mpCStick;
-    /* 0x014 */ mDoDvdThd_mountArchive_c* field_0x14;
+    /* 0x014 */ mDoDvdThd_mountArchive_c* dvd_command_ptr;
```
Comment (TakaRikka) [inline include/d/menu/d_menu_fishing.h:44]: have this member name follow naming guidelines pls

File: include/d/meter/d_meter_HIO.h [hunk 2]
```diff
@@ -184,13 +184,10 @@ class dMeter_drawFishingHIO_c {
...
-    /* 0x200 */ f32 mMaxSizeScale;
-    /* 0x204 */ f32 mCaughtNumScale;
-    /* 0x208 */ f32 mMaxSizePosX;
-    /* 0x20C */ f32 mCaughtNumPosX;
+    /* 0x200 */ f32 mMaxSizeScale[2]; // Probably a different name originally
+    /* 0x208 */ f32 mMaxSizePosX[2];  // Probably a different name originally
     /* 0x210 */ f32 mCaughtNumPosX_4x3;
-    /* 0x214 */ f32 mMaxSizePosY;
-    /* 0x218 */ f32 mCaughtNumPosY;
+    /* 0x214 */ f32 mMaxSizePosY[2]; // Probably a different name originally
```
Comment (TakaRikka) [inline include/d/meter/d_meter_HIO.h:190]: for these, if the names that were there before could be put in the comment that'd be great. like
```cpp
/* 0x200 */ f32 mScale[2];  // 0: MaxSizeScale, 1: CaughtNumScale
```

**Review by TakaRikka — APPROVED** [review]
looks mostly good, just few minor things. also progress conflict needs to be fixed

File: include/d/menu/d_menu_fishing.h (hunk 1 above)
Comment (Trueffeloot) [reply include/d/menu/d_menu_fishing.h:44]: sure this was saved in ghidra like this for some reason but I'll change it

**Comment by Trueffeloot** [comment]
This is good to be merged now👍

**Pull request merged**